feat: option to setInteractionResult without calling interactionFinished

JSantha opened the pull request:
As already mentioned here #350. We have no clear way to propagate interaction result without causing response redirect. To allow custom response handling I'm submitting this non-breaking PR which simply extracts result propagation to its own method.

Now, instead of calling `provider. interactionFinished` we have option to call `provider.setInteractionResult` with same arguments to prevent immediate redirection.

panva commented:
Hi @JSantha and thank you for this PR.

Is returning the interaction object the right call here? Shouldn't just the target url be returned instead?

Can you inline doc mark `setInteractionResult`, `interactionFinished`, `interactionDetails` and `setProviderSession` as public like so?

```
  /**
   * @name setProviderSession
   * @api public
   */
```

panva commented on lib/provider.js:
```diff
@@ -197,27 +197,43 @@ class Provider extends events.EventEmitter {
   }
 
+  /**
+   * @name interactionDetails
```
should be `setProviderSession`

panva commented on docs/configuration.md:
````diff
@@ -320,6 +320,27 @@ router.post('/interaction/:grant', async (ctx, next) => {
 ```
 
+**`#provider.interactionResult`**
+Unlike `#provider.interactionFinished` redirect uri is returned instead of immediate http redirect.
````
replace `redirect uri` with `authorization request resume uri`

JSantha commented:
You are right with returning whole interaction, I've changed that to return `redirectTo` only. Renamed `setInteractionResult` to `interactionResult` as well to make naming more convenient.

panva commented:
Thank you @JSantha
